### PR TITLE
Update Trame Examples: Use Vuetify 3 components

### LINKED ISSUE #5430: Simple Trame App Example not Working
### Describe the bug, what's wrong, and what you expected.

This may be a documentation issue or user error as a total Vue/Vuetify beginner, but the [pyvista trame example](https://tutorial.pyvista.org/tutorial/09_trame/a_trame_simple.html) no longer works for me since updating Vue on my system such that the default is vue3 instead of vue2. I've tried modifying the example to get the server to run, but then I run into a new issue where there is a JS Error. Screenshots and code below.

### Steps to reproduce the bug.

a_trame_simple.py (as currently provided in the pyVista tutorial)
```python
import pyvista as pv
from pyvista import examples
from pyvista.trame.ui import plotter_ui
from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout

pv.OFF_SCREEN = True

server = get_server()
state, ctrl = server.state, server.controller

mesh = examples.load_random_hills()

pl = pv.Plotter()
pl.add_mesh(mesh)

with SinglePageLayout(server) as layout:
    with layout.content:
        view = plotter_ui(pl)

server.start()
```
This results in the following error: 
> TypeError: Server using client_type='vue3' while we expect 'vue2'

previously (on pyvista v 0.42.3) this example ran as expected.

now, if I modify the above example to
`from trame.ui.vuetify import SinglePageLayout`
The server does start! but it does not look as expected.
Here is the resulting screen:
Screen Shot 2024-01-02 at 9 19 44 PM
And a snippet of the errorlog (where the JS Error prints continuously until killed):

> python snippets/a_trame_simple.py

> And to prevent your browser from opening, add '--server' to your command line.
> 
> App running at:
>  - Local:   http://localhost:8080/
>  - Network: http://127.0.0.1:8080/
> 
> Note that for multi-users you need to use and configure a launcher.
>  JS Error => Lookup error: Vue.use(Vuetify.createVuetify(), {})
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => error: Uncaught TypeError: Cannot read properties of null (reading '0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Tue Jan 02 21:26:12 2024 EST

                OS : Darwin
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
               RAM : 64.0 GiB
       Environment : Python
       File system : apfs
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro 5500M OpenGL Engine
       GPU Version : 4.1 ATI-4.6.21
  MathText Support : True

  Python 3.9.17 (main, Jun 20 2023, 17:03:39)  [Clang 13.0.0
  (clang-1300.0.29.30)]

           pyvista : 0.43.1
               vtk : 9.3.0
             numpy : 1.23.3
        matplotlib : 3.7.2
            scooby : 0.7.4
             pooch : v1.6.0
            pillow : 9.2.0
           IPython : 8.5.0
        ipywidgets : 8.1.1
             scipy : 1.9.1
              tqdm : 4.64.1
             trame : 3.5.0
      trame_client : 2.14.2
      trame_server : 2.14.0
         trame_vtk : 2.6.3
     trame_vuetify : 2.3.1
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by annehaley**
### Overview

Resolves #5430.

Trame now defaults to a Vue 3 client, so the current examples in `examples_trame` fail with the following error:
```
TypeError: Server using client_type='vue3' while we expect 'vue2'
```
This is because the component classes in these examples come from Vuetify 2, which expect a Vue 2 -based client. This PR updates these examples to use component classes from Vuetify3. 

### Details

The changes in this PR include the following:
1. Change imports of `vuetify` to `vuetify3` (from both `trame.ui` and `trame.widgets`)
2. Change arguments to Vuetify 3 components where there are API discrepancies between Vuetify 2 and 3:
    a. `dense=True` -> `density="compact"`
    b. `on_icon/off_icon` -> `true_icon/false_icon`
    c. Vuetify 3 `FileInput` component always returns list even when `multiple=False`
3. Minor changes to Viewer classes in `pyvista/trame/ui` to accommodate `custom_ui.py` example
    a. Prevent error when `add_menu=False`
    b. Fix menu button spacing when parent component is not a flex container

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
Thanks!

**Comment by tkoyama010** [comment]
@pyvista/kitware Could someone approve this? According to the specification of CODEOWNER, we need approval from the Kitware team to merge.

**Review by alesgenova — APPROVED** [review]
Looks good!

**Pull request merged**
